karapace/kafka_rest_apis/__init__.py

```diff
@@ -477,27 +477,24 @@ async def _maybe_create_async_producer(self) -> AIOKafkaProducer:
 
                 log.info("Creating async producer")
 
-                # Don't retry if creating the SSL context fails, likely a configuration issue with
-                # ciphers or certificate chains
-                ssl_context = create_client_ssl_context(self.config)
-
-                # Don't retry if instantiating the producer fails, likely a configuration error.
-                producer = AIOKafkaProducer(
-                    acks=acks,
-                    bootstrap_servers=self.config["bootstrap_uri"],
-                    compression_type=self.config["producer_compression_type"],
-                    connections_max_idle_ms=self.config["connections_max_idle_ms"],
-                    linger_ms=self.config["producer_linger_ms"],
-                    max_request_size=self.config["producer_max_request_size"],
-                    metadata_max_age_ms=self.config["metadata_max_age_ms"],
-                    security_protocol=self.config["security_protocol"],
-                    ssl_context=ssl_context,
-                    **get_kafka_client_auth_parameters_from_config(self.config),
-                )
-
                 try:
+                    producer = AsyncKafkaProducer(
```

Comment:
suggestion: Now that the instantiation cannot raise `(NoBrokersAvaiable, AuthenticationFailedError)` anymore, let's move it outside the try-block.

Reply:
For now I've only moved starting the thread, instantiating the underlying producer is not in `start` yet, work in progress.